WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationFormViewModel.swift

```diff
@@ -142,10 +142,17 @@ final class BlazeCampaignCreationFormViewModel: ObservableObject {
                 self.targetUrl = targetUrl
                 self.urlParams = urlParams
         }
-    }
+    }()
 
     // For Ad destination purposes
-    private var productURL: String? { product?.permalink }
+    private var productURL: String? {
+        let productLink = product?.permalink
+        if let productLink, let siteURL, productLink.isEmpty {
+            /// fallback to the default product URL {site_url}?post_type=product&p={product_id}
+            return String(format: "%@?post_type=product&p=%d", siteURL, productID)
```

Comment:
TIL about this. Nice. 

Suggestion: Can we make this URL creation logic inside a function as an extension to `Product`? We can keep this extension in this view model file for now. We could also add unit tests.

Reply:
Good point! I moved this logic to a public extension of `Product` in b58cf613407052c17a37e0f6eb4667a779ae4e15 to reuse the logic when necessary. Also added a new test in 2c7585e68814d9096b1f469d0928ff35eaef2fb6 and remove a legacy one in de4dbc2ede784b49da7e12f2c22f5c4ae2be29f3.